### ISSUE TITLE
Uploading MP4 (uppercase file extension) fails

### ISSUE THREAD

**Issue by DecaTec** [Chocobozzz/PeerTube#4152]
**Describe the current behavior**
I'm trying to upload an *.MP4 file to Peertube (uppercase extension). Here I get following error from the logs:
```
error[2.6.2021, 11:09:48] Error in controller.
{
  "err": "SequelizeValidationError: Validation error: \".MP4\" is not a valid extname.\n    at InstanceValidator._validate (/var/www/peertube/versions/peertube-v3.2.1/node_modules/sequelize/lib/instance-validator.js:78:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n    at async InstanceValidator._validateAndRunHooks (/var/www/peertube/versions/peertube-v3.2.1/node_modules/sequelize/lib/instance-validator.js:111:7)\n    at async InstanceValidator.validate (/var/www/peertube/versions/peertube-v3.2.1/node_modules/sequelize/lib/instance-validator.js:93:12)\n    at async VideoFileModel.save (/var/www/peertube/versions/peertube-v3.2.1/node_modules/sequelize/lib/model.js:3877:7)"
}
```

**Steps to reproduce:**

1. Upload *.MP4 file with uppercase file extension.
2. The upload fails.

**Describe the expected behavior**
The upload of any valid file should proceed if it's (technically) a valid video format. This should not depend on the file extension (upper/lower case).

**Additional information**

* PeerTube instance:
  * URL: local
  * version: 3.2.1
  * NodeJS version: 12.22.1
  * Ffmpeg version: 4.2.4

**Comment by BlueChrystal** [comment]
I quickly investigated this, and found that the maps which the extensions are validated against always have lowercase keys. See these lines (the first snippet is for videos, the second is for both image and audio files):
https://github.com/Chocobozzz/PeerTube/blob/27bc95867442c772841fb183a625bbda61dede51/server/initializers/constants.ts#L1008

https://github.com/Chocobozzz/PeerTube/blob/27bc95867442c772841fb183a625bbda61dede51/server/initializers/constants.ts#L488-L489

In all of these lines, uppercase letters are not accounted for.

There are two possible fixes that I can think of: lowercasing the extension provided to the map everywhere, or making the maps have every possible permutation of the extension with lowercase and uppercase letters. Personally, I highly prefer the earlier.

I'd like to open a PR for this, but first I want to know what fix to make.

**Comment by DecaTec** [comment]
Lowercasing the extensions compared to the map (first solution) seems to be the better solution for me.

I ran a quick test and found that this can be changed:
https://github.com/Chocobozzz/PeerTube/blob/27bc95867442c772841fb183a625bbda61dede51/server/helpers/custom-validators/videos.ts#L81-L83

```
return exists(value) && (value === VIDEO_LIVE.EXTENSION || MIMETYPES.VIDEO.EXT_MIMETYPE[value.toLowerCase()] !== undefined)
```

With this implementation I was able to upload my *.MP4 file without problems.

But I'm not familiar with PeerTube development, that's why I cannot provide an PR.
